### ISSUE TITLE
Unveiling the Role of Learning Rate Schedules via Functional Scaling Laws

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#464]
## Links
https://arxiv.org/abs/2509.19189

NeurIPS 2025 as a spotlight paper

## Authors
 Peking University
[Binghui Li](https://arxiv.org/search/cs?searchtype=author&query=Li,+B), [Fengling Chen](https://arxiv.org/search/cs?searchtype=author&query=Chen,+F), [Zixun Huang](https://arxiv.org/search/cs?searchtype=author&query=Huang,+Z), [Lean Wang](https://arxiv.org/search/cs?searchtype=author&query=Wang,+L), [Lei Wu](https://arxiv.org/search/cs?searchtype=author&query=Wu,+L)

### 🧠 まとめ
本論文は、学習率スケジュール（LRS: Learning Rate Schedule）がスケーリング則に与える影響を理論的に解析し、**学習過程の損失動態を関数的に記述する「Functional Scaling Law (FSL)」**を提案した。  
オンラインSGDを確率微分方程式（SDE）でモデル化し、LRSを畳み込み型の関数項として明示的に取り込むことで、異なるスケジュール間の性能差を解析可能にした。  
特に、定常学習率・指数減衰・Warmup-Stable-Decay（WSD）の3種のスケジュールを解析し、WSDが最も高いスケーリング効率を示すことを理論的に説明した。

---

### 1. 背景と問題設定
従来のスケーリング則研究は「最終ステップでの損失（final-step risk）」のみに注目しており、**学習率スケジュールによる動的変化を無視**していた。  
著者らはこの問題を解決するため、**teacher–student kernel regression**設定に基づき、オンラインSGDを**内在時間（intrinsic time）に基づく確率微分方程式（SDE）**として再定式化。

- 従来研究との違い  
  既存研究 ([Bordelon et al. 2024], [Paquette et al. 2024]) は定常学習率のみを扱っていたが、本研究は**一般的なLRS全体を統一的に解析可能な理論枠組み**を構築。

---

### 2. 提案手法：Functional Scaling Law (FSL)
SGDの期待リスクを以下のように3つの要素に分解した**関数型スケーリング則 (FSL)** を導出。

$$
\mathbb{E}[R(\nu_t)] - \frac{1}{2}\sigma^2 
\simeq 
\underbrace{M^{-s\beta}}_{\text{近似誤差}} +
\underbrace{t^{-s}}_{\text{フルバッチGD項}} +
\underbrace{\int_0^t K(t-r)(e(r)+\sigma^2)\gamma_{\phi,b}(r)\,dr}_{\text{ノイズ項}}
$$

（定理4.1）

- **第1項:** モデル容量 \(M\) に依存する近似誤差  
- **第2項:** 内在時間 \(t\) に依存する学習速度（フルバッチGD相当）  
- **第3項:** 学習率スケジュールとバッチサイズが支配するノイズ項（畳み込み型）

📊 *図1（p.6）* では、理論式（FSL）がSGDの損失曲線を高精度に再現する様子を示し、異なるLRS（Cosine, WSD, Cyclic）に対しても高い一致性を確認。

---

#### (1) Constant LRS  
- スケーリング則：

$$
  E_K \sim M^{-s\beta} + (\eta K)^{-s} + \frac{\eta}{B}(\sigma^2 + (\eta K)^{-(2-1/\beta)})
$$

- **学習率とバッチサイズの線形スケーリング則**（linear scaling rule）を理論的に導出。
- **データ制約・計算制約下での最適スケーリング**
  - Data-optimal: \(E_{\text{opt}}\sim D^{-s/(s+1)}\)
  - Compute-optimal: \(E_{\text{opt}}\sim C^{-s\beta/(1+s\beta+\beta)}\)
- 高容量モデルほど計算効率が良いことを理論的に説明。

#### (2) Exponential Decay LRS  
- スケーリング則：

$$
  E_K \sim M^{-s\beta} + T^{-s} + \sigma^2\left[\frac{b}{B}+\frac{(a-b)\min(M,T^{1/\beta})}{BT}\right]
$$

- 指数減衰によりスケーリング指数が向上し、学習効率が改善。  
  → **学習率減衰が有効な理由を理論的に説明。**
- Data-optimal regime: \(E_{\text{opt}} \sim (D/\log D)^{-s\beta/(1+s\beta)}\)  
- Compute-optimal regime: \(E_{\text{opt}} \sim (C/\log C)^{-s\beta/(2+s\beta)}\)

#### (3) Warmup-Stable-Decay (WSD) LRS  
- 2段階構成: 安定期 (stable phase) → 減衰期 (decay phase)
- スケーリング則：

$$
  E_K \sim M^{-s\beta} + (T_1+T_2)^{-s} + \sigma^2\!\left[\frac{b}{B} + \frac{(a-b)\min(M,T_2^{1/\beta})}{BT_2}\right]
$$

- 安定期により**内在時間が増加**し、指数減衰に比べ**対数項の損失を除去**。  
- 特に難しいタスク（hard regime）で顕著な改善を示す。  
- *図2（p.10）* では、WSDが他のスケジュールよりも低い最終リスクに到達する様子を可視化。

---

#### (1) Teacher–Student回帰での検証
- *図2(b)* において、FSL理論と実際のSGD軌跡が一致。  
- WSD > Exponential > Constant の順に性能が向上。

#### (2) LLM Pre-trainingでの応用
- LLaMA (400M / 1B) および QwenMoE (1B) モデルに適用。
- FSLにより**損失曲線の予測とフィッティング**が高精度に再現 (*図3a, b, p.11*)。
- FSLを用いた最適化により、**経験的最良なWSDに類似したLRS**が自動的に導出され、従来より低い最終損失を実現。

---

### 5. これまでの研究との違い・新規性
| 項目 | 既存研究 | 本研究 |
|------|-----------|--------|
| 対象 | 最終損失のスケーリング | 学習過程全体の損失動態 |
| モデル化 | 静的な近似式 | SDEベースの動的モデル |
| LRSの扱い | 定常または簡略化 | 任意のスケジュールを統一的に解析 |
| 新規概念 | なし | **内在時間 (Intrinsic Time)** と**FSL**導入 |
| 貢献 | 理論限界の分析 | 理論＋LLM実験で妥当性を実証 |

---

### 6. 貢献の実証方法
- **理論解析**: kernel regression + SDE解析によるFSL導出  
- **数値実験**: FSLとSGDの損失曲線の整合性検証  
- **LLM実験**: FSLによるLRS設計最適化（LLaMA, GPT, QwenMoE）  
- **Figure 1–3, 5–10** を通じて、理論・数値・実際のLLMでの一致を実証。

---

### 7. 重要な図の説明
- **Figure 1 (p.6):** FSLの3項構造と実際のSGD曲線の一致。  

<img width="999" height="374" alt="Image" src="https://github.com/user-attachments/assets/f15d0f05-a2bf-44ae-bb6c-12611c01214d" />

- **Table 1 (p.2):** 各LRSごとのデータ／計算効率最適スケーリング則を整理。  

<img width="1324" height="335" alt="Image" src="https://github.com/user-attachments/assets/39ff7d5e-d3c9-4422-babd-af24ab418a71" />

- **Figure 2 (p.10):** LRS別のリスク曲線比較。WSDが最も高速かつ安定に収束。  

<img width="1026" height="479" alt="Image" src="https://github.com/user-attachments/assets/ddd30bd6-2a02-44f8-8d0c-8f97cf30e8bb" />

- **Figure 3 (p.11):** LLaMA/QwenMoEでのFSL適用結果。FSL-optimal LRSが最良性能。  

<img width="663" height="276" alt="Image" src="https://github.com/user-attachments/assets/6a056d19-dc80-4f01-954c-07ddc7652d3d" />

- **Figure 5–10 (Appendix):** モデルサイズ・学習ステップ・LRS比率を変えた追加検証。

---

## 🔍 総括
本研究は、**学習率スケジュールとスケーリング則の理論的リンクを初めて厳密に定式化**した点で画期的である。  
Functional Scaling Lawは、LLMのトレーニングダイナミクスを理解・予測・最適化するための新しい解析フレームワークとして機能し、今後の学習率設計理論に大きな指針を与える。
